envi/__init__.py

```diff
@@ -156,6 +156,26 @@ def archGetRegisterGroups(self):
         allr = [rname for rname in regctx.getRegisterNames()]
         return [ ('all', allr), ]
 
+    def archModifyFuncAddr(self, va, arch):
+        '''
+        Returns a potentially modified set of (va, arch).
+        Default: return the same va, arch
+
+        This hook allows an architecture to correct VA and Architecture, such 
+        as is necessary for ARM/Thumb.
+        '''
+        return va, arch
```

Comment:
this is too specific to the arm/thumb case where the only thing it will be doing is changing archs.  we should probably return va,info ( where info is a dict which can have whatever an arch needs returned as context info )